src/axom/klee/GeometryOperatorsIO.cpp

```diff
@@ -0,0 +1,662 @@
+  }
+  return givenOrigin;
+}
+
+/**
+ * Get the normal vector to use for a perpendicular slice.
+ *
+ * \param sliceProxy the Proxy describing the slice
+ * \param defaultNormal the default normal vector
+ * \return the vector to use as the normal
+ */
+primal::Vector3D getPerpendicularSliceNormal(const inlet::Proxy &sliceProxy,
+                                             const primal::Vector3D &defaultNormal)
+{
+  if(!sliceProxy.contains("normal"))
+  {
+    return defaultNormal;
+  }
+
+  primal::Vector3D givenNormal =
+    toVector(sliceProxy, "normal", Dimensions::Three);
+  auto cross = primal::Vector3D::cross_product(givenNormal, defaultNormal);
+  bool parallel = cross.is_zero();
+  if(!parallel)
+  {
+    throw KleeError({sliceProxy["normal"].name(), "Invalid normal"});
+  }
+  return givenNormal;
+}
+
+/**
+ * Read a perpendicular slice.
+ *
+ * \param sliceContainer the Container describing the slice
+ * \param planeName the name of the plane ("x", "y", or "z")
+ * \param defaultNormal the default normal vector for the type of plane
+ *  being parsed
+ * \param defaultUp the default up vector for the plane being parsed
+ * \param startProperties the properties prior to this operator
+ * \return the parsed plane
+ */
+OpPtr readPerpendicularSlice(const inlet::Container &sliceContainer,
+                             char const *planeName,
+                             Vector3D const &defaultNormal,
+                             Vector3D const &defaultUp,
+                             const TransformableGeometryProperties &startProperties)
+{
+  verifyObjectFields(sliceContainer,
+                     planeName,
+                     FieldSet {},
+                     {"origin", "normal", "up"});
+  const primal::Vector3D defaultNormalVec {defaultNormal.data()};
+
+  auto origin =
+    getPerpendicularSliceOrigin(sliceContainer, planeName, defaultNormalVec);
+  auto normal = getPerpendicularSliceNormal(sliceContainer, defaultNormalVec);
+  auto up = toVector(sliceContainer, "up", Dimensions::Three, defaultUp);
+
+  return makeCheckedSlice(origin,
+                          normal,
+                          up,
+                          startProperties,
+                          sliceContainer.name());
+}
+
+/**
+ * Parse a "slice" operator.
+ *
+ * \param opContainer the Container from which to read the operator
+ * \param startProperties the properties prior to this operator
+ * \return the created operator
+ */
+OpPtr parseSlice(const inlet::Container &opContainer,
+                 const TransformableGeometryProperties &startProperties)
+{
+  if(startProperties.dimensions != Dimensions::Three)
+  {
+    throw KleeError({opContainer.name(), "Cannot do a slice from 2D"});
+  }
+  verifyObjectFields(opContainer, "slice", FieldSet {}, FieldSet {});
+  auto &sliceContainer =
+    *opContainer.getChildContainers().at(opContainer.name() + "/slice").get();
+  inlet::Proxy sliceProxy {sliceContainer};
```

Comment:
I may be missing something, but can this function (and toPoint/toVector and all the way down the call stack) just operate on an `inlet::Container` instead of an `inlet::Proxy`?  The methods that get used on this proxy object (`contains()` and `get<std::vector<T>>()`) should forward directly to identical methods on the underlying `Container` but in my opinion it would be a little more clear to use `Container` where possible instead of a `Proxy` - which could also be a `Field` or `Function`